Fishamble feature #168 report posts/comments

John-Paul-Larkin opened the pull request:
# ✨ Codu Pull Request 💻

![Codu Logo](https://raw.githubusercontent.com/codu-code/codu/develop/public/images/codu-gradient.png)

## Pull Request details:
Added a flag icon to comments, which opens a modal allowing users to report the comment. Once submitted, the report and details should be sent to an admin.
Just need to add an ADMIN_EMAIL to the process.env for the recipient who will recieve the reports.


## Any Breaking changes:
none

## Associated Screenshots:
https://youtu.be/tIwlSFcrehI
![codu_reportjpg](https://github.com/codu-code/codu/assets/107807466/4d9e8c5f-f68e-44c7-aee9-652d19db032c)

xiaoniuniu89 committed:
fix(softbreak): override OOTB softbreak to render as a newline - #191 (#194)

xiaoniuniu89 committed:
fix(media tags): fix for aspect ratio bug caused by overriding style tags - process custom tag links if not correct format #189 (#196)

NiallJoeMaher committed:
Metadata for user profile (#200)

lmammino committed:
Added missing protocol in og:url meta tag for Articles (#202)

xiaoniuniu89 committed:
Dacallaghan/editor shortcuts (#201)

* feat(shortcuts): add hotkeys and shortcuts for the blog editor

NiallJoeMaher committed:
Simple filter options for articles (#205)

* Simple filter options for articles

xiaoniuniu89 committed:
fix(security):update next-auth to latest which includes security patch

NiallJoeMaher committed:
Link Avatars to profiles

NiallJoeMaher committed:
Enable filter by tags (#212)

Tags can be clicked and will show posts that are related to that tag.

NiallJoeMaher committed:
Feat/welcome-email (#215)

* Update Prisma, add SES and Nodemailer
* Remove sign-up tracking (it's not working)
* Send an email when user signs up
* Discord invite via .env

NiallJoeMaher committed:
Remove getCount while no session (#217)

Also clean up error handling and remove await from sendmail.

NiallJoeMaher committed:
Feat/223 - Add alpha pages (#224)

* Add editor to alpha folder
* Temp removal of welcome email

snyk-bot committed:
fix: upgrade @aws-sdk/s3-request-presigner from 3.235.0 to 3.289.0 (#222)

Snyk has created this PR to upgrade @aws-sdk/s3-request-presigner from 3.235.0 to 3.289.0.

See this package in npm:
https://www.npmjs.com/package/@aws-sdk/s3-request-presigner

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

snyk-bot committed:
fix: upgrade @aws-sdk/client-s3 from 3.235.0 to 3.289.0 (#221)

Snyk has created this PR to upgrade @aws-sdk/client-s3 from 3.235.0 to 3.289.0.

See this package in npm:
https://www.npmjs.com/package/@aws-sdk/client-s3

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

snyk-bot committed:
fix: upgrade react-hook-form from 7.34.0 to 7.43.5 (#220)

Snyk has created this PR to upgrade react-hook-form from 7.34.0 to 7.43.5.

See this package in npm:
https://www.npmjs.com/package/react-hook-form

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

snyk-bot committed:
fix: upgrade @next/bundle-analyzer from 12.2.3 to 12.3.4 (#219)

Snyk has created this PR to upgrade @next/bundle-analyzer from 12.2.3 to 12.3.4.

See this package in npm:
https://www.npmjs.com/package/@next/bundle-analyzer

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

NiallJoeMaher committed:
fix: upgrade constructs from 10.1.210 to 10.1.277 (#225)

Snyk has created this PR to upgrade constructs from 10.1.210 to 10.1.277.

See this package in npm:
https://www.npmjs.com/package/constructs

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/0c10c5ab-78ba-42aa-b499-d828ca03741c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

snyk-bot committed:
fix: upgrade @trpc/client from 10.4.1 to 10.16.0 (#226)

Snyk has created this PR to upgrade @trpc/client from 10.4.1 to 10.16.0.

See this package in npm:
https://www.npmjs.com/package/@trpc/client

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

snyk-bot committed:
fix: upgrade @trpc/server from 10.4.1 to 10.16.0 (#229)

Snyk has created this PR to upgrade @trpc/server from 10.4.1 to 10.16.0.

See this package in npm:
https://www.npmjs.com/package/@trpc/server

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

snyk-bot committed:
fix: upgrade @trpc/next from 10.4.1 to 10.16.0 (#228)

Snyk has created this PR to upgrade @trpc/next from 10.4.1 to 10.16.0.

See this package in npm:
https://www.npmjs.com/package/@trpc/next

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: Niall Maher <niall@codu.ie>

snyk-bot committed:
fix: upgrade @trpc/react-query from 10.4.1 to 10.16.0 (#227)

Snyk has created this PR to upgrade @trpc/react-query from 10.4.1 to 10.16.0.

See this package in npm:
https://www.npmjs.com/package/@trpc/react-query

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: Niall Maher <niall@codu.ie>

NiallJoeMaher committed:
Uncomment welcome email event (#230)

NiallJoeMaher committed:
Allow long words or urls to wrap (#233)

NiallJoeMaher committed:
fix: upgrade @tanstack/react-query from 4.18.0 to 4.27.0 (#234)

Snyk has created this PR to upgrade @tanstack/react-query from 4.18.0 to 4.27.0.

See this package in npm:
https://www.npmjs.com/package/@tanstack/react-query

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

snyk-bot committed:
fix: upgrade constructs from 10.1.277 to 10.1.280 (#238)

Snyk has created this PR to upgrade constructs from 10.1.277 to 10.1.280.

See this package in npm:
https://www.npmjs.com/package/constructs

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/0c10c5ab-78ba-42aa-b499-d828ca03741c?utm_source=github&utm_medium=referral&page=upgrade-pr

NiallJoeMaher committed:
fix: upgrade satori from 0.1.2 to 0.4.3 (#239)

Snyk has created this PR to upgrade satori from 0.1.2 to 0.4.3.

See this package in npm:
https://www.npmjs.com/package/satori

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
fix: upgrade @headlessui/react from 1.6.6 to 1.7.13 (#237)

Snyk has created this PR to upgrade @headlessui/react from 1.6.6 to 1.7.13.

See this package in npm:
https://www.npmjs.com/package/@headlessui/react

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

snyk-bot committed:
fix: upgrade superjson from 1.9.1 to 1.12.2 (#241)

Snyk has created this PR to upgrade superjson from 1.9.1 to 1.12.2.

See this package in npm:
https://www.npmjs.com/package/superjson

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

snyk-bot committed:
fix: upgrade constructs from 10.1.280 to 10.1.282 (#240)

Snyk has created this PR to upgrade constructs from 10.1.280 to 10.1.282.

See this package in npm:
https://www.npmjs.com/package/constructs

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/0c10c5ab-78ba-42aa-b499-d828ca03741c?utm_source=github&utm_medium=referral&page=upgrade-pr

snyk-bot committed:
fix: upgrade @aws-sdk/client-s3 from 3.289.0 to 3.295.0 (#242)

Snyk has created this PR to upgrade @aws-sdk/client-s3 from 3.289.0 to 3.295.0.

See this package in npm:
https://www.npmjs.com/package/@aws-sdk/client-s3

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

NiallJoeMaher committed:
Downgrade compute size on prod

NiallJoeMaher committed:
Test cloudfront distribution on dev

snyk-bot committed:
fix: upgrade constructs from 10.1.282 to 10.1.285 (#244)

Snyk has created this PR to upgrade constructs from 10.1.282 to 10.1.285.

See this package in npm:
https://www.npmjs.com/package/constructs

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/0c10c5ab-78ba-42aa-b499-d828ca03741c?utm_source=github&utm_medium=referral&page=upgrade-pr

NiallJoeMaher committed:
Web CDN with CacheCookieBehavior.all()

NiallJoeMaher committed:
Use web distribution and add cookie settings

NiallJoeMaher committed:
Add default object to web distribution

NiallJoeMaher committed:
Update headers and ttl - cache options

NiallJoeMaher committed:
fix: upgrade @resvg/resvg-js from 2.2.0 to 2.4.1 (#245)

Snyk has created this PR to upgrade @resvg/resvg-js from 2.2.0 to 2.4.1.

See this package in npm:
https://www.npmjs.com/package/@resvg/resvg-js

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
CloudFront-Viewer-Country to headers

NiallJoeMaher committed:
Update app ID

NiallJoeMaher committed:
Revert ecs naming

NiallJoeMaher committed:
fix: upgrade @aws-sdk/client-s3 from 3.295.0 to 3.298.0 (#246)

Snyk has created this PR to upgrade @aws-sdk/client-s3 from 3.295.0 to 3.298.0.

See this package in npm:
https://www.npmjs.com/package/@aws-sdk/client-s3

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

snyk-bot committed:
fix: upgrade constructs from 10.1.285 to 10.1.290 (#247)

Snyk has created this PR to upgrade constructs from 10.1.285 to 10.1.290.

See this package in npm:
https://www.npmjs.com/package/constructs

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/0c10c5ab-78ba-42aa-b499-d828ca03741c?utm_source=github&utm_medium=referral&page=upgrade-pr

NiallJoeMaher committed:
Forward host listner rule

NiallJoeMaher committed:
Add default listener of not allowed

NiallJoeMaher committed:
Remove commented code

snyk-bot committed:
fix: upgrade @markdoc/markdoc from 0.1.13 to 0.2.2 (#249)

Snyk has created this PR to upgrade @markdoc/markdoc from 0.1.13 to 0.2.2.

See this package in npm:
https://www.npmjs.com/package/@markdoc/markdoc

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

NiallJoeMaher committed:
Upgrade packages suggested by Snyk (#248)

* Upgrade packages suggested by Snyk
* Upgrade and install packages with node @LTS
* Downgrade Next version from recommended version (breaks build if using current)

NiallJoeMaher committed:
Bugfix/revert node v (#250)

* Revert node versions to 16 for build

* Add overflow scroll to comments

NiallJoeMaher committed:
fix: upgrade constructs from 10.1.290 to 10.1.293 (#252)

Snyk has created this PR to upgrade constructs from 10.1.290 to 10.1.293.

See this package in npm:
https://www.npmjs.com/package/constructs

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/0c10c5ab-78ba-42aa-b499-d828ca03741c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
Feat - Consistent grey across the site (#253)

* Update Tailwind config with new black

* Update globals to use neutral gray

* Remove slate, gray and smoke colors

* Remove quotes from blockquote

NiallJoeMaher committed:
Remove white border from nav (#255)

NiallJoeMaher committed:
fix: upgrade constructs from 10.1.293 to 10.1.296 (#256)

Snyk has created this PR to upgrade constructs from 10.1.293 to 10.1.296.

See this package in npm:
https://www.npmjs.com/package/constructs

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/0c10c5ab-78ba-42aa-b499-d828ca03741c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
Fix margin header border width (#258)

NiallJoeMaher committed:
Feat/admin view (#260)

* Bump Prisma versions
* Simple metrics page for ADMINS
* Breakpoints for mobile

NiallJoeMaher committed:
fix: upgrade constructs from 10.1.296 to 10.1.298 (#261)

Snyk has created this PR to upgrade constructs from 10.1.296 to 10.1.298.

See this package in npm:
https://www.npmjs.com/package/constructs

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/0c10c5ab-78ba-42aa-b499-d828ca03741c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

snyk-bot committed:
fix: upgrade constructs from 10.1.298 to 10.1.299 (#263)

Snyk has created this PR to upgrade constructs from 10.1.298 to 10.1.299.

See this package in npm:
https://www.npmjs.com/package/constructs

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/0c10c5ab-78ba-42aa-b499-d828ca03741c?utm_source=github&utm_medium=referral&page=upgrade-pr

snyk-bot committed:
fix: upgrade constructs from 10.1.299 to 10.1.300 (#264)

Snyk has created this PR to upgrade constructs from 10.1.299 to 10.1.300.

See this package in npm:
https://www.npmjs.com/package/constructs

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/0c10c5ab-78ba-42aa-b499-d828ca03741c?utm_source=github&utm_medium=referral&page=upgrade-pr

snyk-bot committed:
fix: upgrade constructs from 10.1.300 to 10.1.302 (#266)

Snyk has created this PR to upgrade constructs from 10.1.300 to 10.1.302.

See this package in npm:
https://www.npmjs.com/package/constructs

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/0c10c5ab-78ba-42aa-b499-d828ca03741c?utm_source=github&utm_medium=referral&page=upgrade-pr

NiallJoeMaher committed:
Configure natGateways to 1

NiallJoeMaher committed:
Remove extra VPC settings

NiallJoeMaher committed:
Adds desktop version of suggested tags (#267)

NiallJoeMaher committed:
fix: upgrade constructs from 10.1.302 to 10.1.304 (#268)

NiallJoeMaher committed:
fix: upgrade constructs from 10.1.304 to 10.1.305 (#270)

Snyk has created this PR to upgrade constructs from 10.1.304 to 10.1.305.

See this package in npm:
https://www.npmjs.com/package/constructs

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/0c10c5ab-78ba-42aa-b499-d828ca03741c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
Page wrapper changed to mx-2

snyk-bot committed:
fix: upgrade constructs from 10.1.305 to 10.1.306 (#271)

patrickhladun committed:
Post editor (#262)

* Add post editor sticky header and preview icon button
* Remove toaster on save event
* Create editor specific layout
* Add EditorHints component to display the article editor hints and tips
* Use EditorHints and LayoutEditor components

NiallJoeMaher committed:
Test lambda@edge

NiallJoeMaher committed:
fix: upgrade constructs from 10.1.306 to 10.1.308 (#273)

NiallJoeMaher committed:
fix: upgrade constructs from 10.1.308 to 10.1.309 (#275)

Snyk has created this PR to upgrade constructs from 10.1.308 to 10.1.309.

See this package in npm:
https://www.npmjs.com/package/constructs

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/0c10c5ab-78ba-42aa-b499-d828ca03741c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
Fix tsc build issues, replace Vercel OG

NiallJoeMaher committed:
Update CDK dependencies (#276)

NiallJoeMaher committed:
Add @types/react

NiallJoeMaher committed:
Add sharp

JohnAllenTech committed:
Feat/add announcement to articles page (#277)

* feat: add announcement section to articles page

NiallJoeMaher committed:
Bugfix/code cleanup (#278)

* Cleanup CDK, remove Edge Stack from build

* Cleanup announcement banner

* Descriptive alt text

NiallJoeMaher committed:
Remove Edge Stack

NiallJoeMaher committed:
Rename VPC id

NiallJoeMaher committed:
Add second VPC to allow swaping of VPC

NiallJoeMaher committed:
Revert last commit

sleithdylan committed:
Copy link option (#279)

* feat: add copy to clipboard on post

* fix: not redirecting to social media platforms

JohnAllenTech committed:
fix: excerpt now generated from body if not provided (#280)

* fix: excerpt now generated from body if not provided

viniirbr committed:
Feat/new sponsorship (#281)

* feat: add sponsorship page images

* feat: put images

* feat: responsivity

* feat: rotate pictures effect

* Refactor images list code and add alt message

JohnAllenTech committed:
feat: added new props to prompt service and updated style to match site (#282)

Co-authored-by: John Allen <john.allen.ireland@gmail.com>

JohnAllenTech committed:
chore: added canonical link if its in the article (#284)

* chore: added canonical link if its in the article

MartiinWalsh committed:
Open pdf on button press (#288)

snyk-bot committed:
fix: upgrade constructs from 10.2.16 to 10.2.17 (#290)

sleithdylan committed:
fix: copied state for copy to clipboard option (#294)

CBID2 committed:
Create greetings.yml (#295)

snyk-bot committed:
fix: upgrade satori from 0.7.4 to 0.8.0 (#297)

Snyk has created this PR to upgrade satori from 0.7.4 to 0.8.0.

See this package in npm:
https://www.npmjs.com/package/satori

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/0c10c5ab-78ba-42aa-b499-d828ca03741c?utm_source=github&utm_medium=referral&page=upgrade-pr

snyk-bot committed:
fix: upgrade constructs from 10.2.17 to 10.2.19 (#296)

Snyk has created this PR to upgrade constructs from 10.2.17 to 10.2.19.

See this package in npm:
https://www.npmjs.com/package/constructs

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/0c10c5ab-78ba-42aa-b499-d828ca03741c?utm_source=github&utm_medium=referral&page=upgrade-pr

snyk-bot committed:
fix: upgrade satori from 0.8.0 to 0.8.1 (#300)

Snyk has created this PR to upgrade satori from 0.8.0 to 0.8.1.

See this package in npm:
https://www.npmjs.com/package/satori

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/0c10c5ab-78ba-42aa-b499-d828ca03741c?utm_source=github&utm_medium=referral&page=upgrade-pr

snyk-bot committed:
fix: upgrade constructs from 10.2.19 to 10.2.22 (#299)

Snyk has created this PR to upgrade constructs from 10.2.19 to 10.2.22.

See this package in npm:
https://www.npmjs.com/package/constructs

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/0c10c5ab-78ba-42aa-b499-d828ca03741c?utm_source=github&utm_medium=referral&page=upgrade-pr

NiallJoeMaher committed:
fix: upgrade constructs from 10.2.22 to 10.2.24 (#302)

Snyk has created this PR to upgrade constructs from 10.2.22 to 10.2.24.

See this package in npm:
https://www.npmjs.com/package/constructs

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/0c10c5ab-78ba-42aa-b499-d828ca03741c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

snyk-bot committed:
fix: upgrade constructs from 10.2.24 to 10.2.26 (#303)

Snyk has created this PR to upgrade constructs from 10.2.24 to 10.2.26.

See this package in npm:
https://www.npmjs.com/package/constructs

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/0c10c5ab-78ba-42aa-b499-d828ca03741c?utm_source=github&utm_medium=referral&page=upgrade-pr

NiallJoeMaher committed:
fix: upgrade constructs from 10.2.26 to 10.2.27 (#304)

Snyk has created this PR to upgrade constructs from 10.2.26 to 10.2.27.

See this package in npm:
https://www.npmjs.com/package/constructs

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/0c10c5ab-78ba-42aa-b499-d828ca03741c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

kera-cudmore committed:
add: no index meta tag to sponsorship alpha page (#308)

CBID2 committed:
[feat] Create security policy (#306)

NiallJoeMaher committed:
fix: upgrade constructs from 10.2.27 to 10.2.31 (#307)

Snyk has created this PR to upgrade constructs from 10.2.27 to 10.2.31.

See this package in npm:
https://www.npmjs.com/package/constructs

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/0c10c5ab-78ba-42aa-b499-d828ca03741c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

serzhan181 committed:
feat: posts count on profile page (#313)

xiaoniuniu89 committed:
Dacallaghan/slate editor (#312)

* add slate editor fromo example, add material ui

* add slate-markdown, plate  and unified libraries - add serializer for makdown to slate

* add hoverable toolbar
update onchange event to work with slate editor

* update styles for Editable

* switch to html based approach for serialization and deserilalization
- add slate-serializers pacage
- add DomPurify to sanitize html
- html-react-parser to render html in react componant

* fix underline bug

* update to Plate editor and fix initial value on render of editor

* update hoovering toolbar to plate implementation

* add link funcitonality, add support for empty p tags

* add custom font size plugin

* remove html to slate deserializer

* add h3 and h4 heading toolbar buttons

* add expanding toolbar skeleton

* add support for basic embed and images

* update open menu icon and menu position

* add plugin support for lists and keyboard shortcut highlighting

* overriding plate styles for link and keyboard shortcut

* add styles for code block plugin

* add newline after blockquote and code block injection

* add newline break for media embed and img nodes

* testing plate ootb serializer

* add style for heading plugins

* update heading styles

* add basic serialization for code blocks

* add deserealiozation rule for code blocks

* add serialization for img tags

* fix html deserializer bug for img tags creating malformed children arrays in slate object

* fix caption text area resizing on load

* fix slate ol and ul ::marker element bug

* add plate component for code sandbpx and codepen, add logic to parse and prepare media links

* add fix for code sandbox and codepen padding issue

* add serialization rule for media embed - slate to html

* add rule for media_embed html to slate

* add prism highlighting to code block

* chore(organize folder structure): clean up. file and folder configuration

* add parse html functionality to blog view

* chore(typesafety): fixing typescript errors

* chore(cleanup): removing unused code and adding typechecks

* chore(cleanup): cleanup unused code and add some typesafety

* add comments

* update styles for p tags in plate editor

* add style for a tag

* add style for blockquote

* add style for img caption

* add style for lists

* chore(cleanup): removing unused npm packages

* remove console logs

* fix linting errors

NiallJoeMaher committed:
fix: upgrade constructs from 10.2.31 to 10.2.34 (#314)

Snyk has created this PR to upgrade constructs from 10.2.31 to 10.2.34.

See this package in npm:
https://www.npmjs.com/package/constructs

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/0c10c5ab-78ba-42aa-b499-d828ca03741c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
fix: upgrade satori from 0.8.1 to 0.9.1 (#315)

Snyk has created this PR to upgrade satori from 0.8.1 to 0.9.1.

See this package in npm:
https://www.npmjs.com/package/satori

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/0c10c5ab-78ba-42aa-b499-d828ca03741c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

viniirbr committed:
Show user website url (#316)

NiallJoeMaher committed:
fix: upgrade constructs from 10.2.34 to 10.2.36 (#317)

NiallJoeMaher committed:
fix: cdk/package.json & cdk/package-lock.json to reduce vulnerabilities (#318)

NiallJoeMaher committed:
fix: upgrade constructs from 10.2.36 to 10.2.38 (#319)

NiallJoeMaher committed:
fix: package.json & package-lock.json to reduce vulnerabilities (#330)

sonngdev committed:
Improve images on sponsorship page, add padding to notification page title (#331)

NiallJoeMaher committed:
Hero for new homepage build (#289)

* Add first draft of hero for new homepage build

* Update config

* Add alt tags and responsiveness to space scene

* Draft of CTA section

* Add trending posts section

* Add bookmarks to hompage and noindex

sonngdev committed:
Create Courses page (#335)

* feat: create /alpha/courses page

* feat: add course preview

* feat: add responsiveness

* feat: use Next Image for CoursePreview

NiallJoeMaher committed:
Move new editor to /alpha (#343)

viniirbr committed:
Turn off comments on certain posts/articles (#332)

* add disable comments option

* Change name of field allowComments to showComments

serzhan181 committed:
Feat/ Course Overview Page (#342)

* feat: course contents(mock)
* feat: add a progress bar

NiallJoeMaher committed:
Dependency upgrades (#345)

* Remove unused console.log

* Batch update of dependencies

NiallJoeMaher committed:
fix: upgrade slate-react from 0.94.2 to 0.97.1 (#347)

Snyk has created this PR to upgrade slate-react from 0.94.2 to 0.97.1.

See this package in npm:
https://www.npmjs.com/package/slate-react

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
fix: upgrade satori from 0.4.14 to 0.10.1 (#348)

Snyk has created this PR to upgrade satori from 0.4.14 to 0.10.1.

See this package in npm:
https://www.npmjs.com/package/satori

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
fix: upgrade satori from 0.9.1 to 0.10.1 (#353)

Snyk has created this PR to upgrade satori from 0.9.1 to 0.10.1.

See this package in npm:
https://www.npmjs.com/package/satori

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/0c10c5ab-78ba-42aa-b499-d828ca03741c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
fix: upgrade @markdoc/markdoc from 0.2.2 to 0.3.0 (#351)

Snyk has created this PR to upgrade @markdoc/markdoc from 0.2.2 to 0.3.0.

See this package in npm:
https://www.npmjs.com/package/@markdoc/markdoc

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

Shane-Donlon committed:
#344 Show copy code block icon only on hover (#352)

* Feat update: button moved for DOM order

* Style update: Grid add, button positioned

* Style update: text-wrap add for responsivness

* Feat update: tabindex add for accessibility

* Style add: button style for hover, and focus states

* Script remove: unnecessary declaration as now in grid

* Style update: border styles update to parent code

* Style update: media query add for button, 'pre' margin and padding update

* Style update: margins and padding update on .code and 'pre'

* Style update: focus visible changed to focus within to keep button scale when tabbing on code excerpt

* Style update: prefers light mode add

* Style update: light mode white changed to off-white

* Style update: 'black' updated to reflect branding colours

* Style update: codeblock off-black colour update

NiallJoeMaher committed:
Release new homepage, ban users, unban users (#356)

* Upgrade vulnerable packages

* Lighten the font of the count on profile

* Finish first version of new home

* Turn off Slate editor while broken

* Turn off slate editor while broken

* remove unecessary migration

* Paste from alpha homepage

* bugfixes for homepage + cleanup

* Add role to session

* Remove unused imports

* Add ability to delete post as admin

* Make trending procedure public

* Add ability to ban/unban user

NiallJoeMaher committed:
Alter hero layout for mobile exp (#358)

NiallJoeMaher committed:
Add max-h with SVH unit

NiallJoeMaher committed:
fix: upgrade sharp from 0.32.1 to 0.32.2 (#359)

Snyk has created this PR to upgrade sharp from 0.32.1 to 0.32.2.

See this package in npm:
https://www.npmjs.com/package/sharp

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/0c10c5ab-78ba-42aa-b499-d828ca03741c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
fix: upgrade @aws-sdk/client-s3 from 3.367.0 to 3.369.0 (#360)

Snyk has created this PR to upgrade @aws-sdk/client-s3 from 3.367.0 to 3.369.0.

See this package in npm:
https://www.npmjs.com/package/@aws-sdk/client-s3

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
fix: upgrade dompurify from 3.0.4 to 3.0.5 (#363)

Snyk has created this PR to upgrade dompurify from 3.0.4 to 3.0.5.

See this package in npm:
https://www.npmjs.com/package/dompurify

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
Update metadata for SEO

NiallJoeMaher committed:
fix: upgrade sharp from 0.32.2 to 0.32.3 (#365)

Snyk has created this PR to upgrade sharp from 0.32.2 to 0.32.3.

See this package in npm:
https://www.npmjs.com/package/sharp

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/0c10c5ab-78ba-42aa-b499-d828ca03741c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
fix: upgrade slate-react from 0.97.1 to 0.97.2 (#364)

Snyk has created this PR to upgrade slate-react from 0.97.1 to 0.97.2.

See this package in npm:
https://www.npmjs.com/package/slate-react

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
fix: upgrade @aws-sdk/s3-request-presigner from 3.367.0 to 3.369.0 (#362)

Snyk has created this PR to upgrade @aws-sdk/s3-request-presigner from 3.367.0 to 3.369.0.

See this package in npm:
https://www.npmjs.com/package/@aws-sdk/s3-request-presigner

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
Remove no index tag

sleithdylan committed:
fix: share buttons not working on articles (#371)

NiallJoeMaher committed:
fix: upgrade @aws-sdk/client-s3 from 3.369.0 to 3.370.0 (#369)

Snyk has created this PR to upgrade @aws-sdk/client-s3 from 3.369.0 to 3.370.0.

See this package in npm:
https://www.npmjs.com/package/@aws-sdk/client-s3

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
fix: upgrade @aws-sdk/s3-request-presigner from 3.369.0 to 3.370.0 (#368)

Snyk has created this PR to upgrade @aws-sdk/s3-request-presigner from 3.369.0 to 3.370.0.

See this package in npm:
https://www.npmjs.com/package/@aws-sdk/s3-request-presigner

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
fix: upgrade next-auth from 4.22.1 to 4.22.3 (#372)

Snyk has created this PR to upgrade next-auth from 4.22.1 to 4.22.3.

See this package in npm:
https://www.npmjs.com/package/next-auth

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
fix: upgrade aws-cdk-lib from 2.87.0 to 2.88.0 (#374)

Snyk has created this PR to upgrade aws-cdk-lib from 2.87.0 to 2.88.0.

See this package in npm:
https://www.npmjs.com/package/aws-cdk-lib

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/0c10c5ab-78ba-42aa-b499-d828ca03741c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
fix: upgrade superjson from 1.12.4 to 1.13.1 (#373)

Snyk has created this PR to upgrade superjson from 1.12.4 to 1.13.1.

See this package in npm:
https://www.npmjs.com/package/superjson

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
fix: upgrade @aws-sdk/client-s3 from 3.370.0 to 3.374.0 (#376)

Snyk has created this PR to upgrade @aws-sdk/client-s3 from 3.370.0 to 3.374.0.

See this package in npm:
https://www.npmjs.com/package/@aws-sdk/client-s3

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
Bugfix/fix code blocks (#377)

* Update older packages
* Fix code blocks

NiallJoeMaher committed:
Fix width of articles on smaller screens

NiallJoeMaher committed:
Update CDK dependencies

NiallJoeMaher committed:
Update Node to v18

NiallJoeMaher committed:
Fix broken Dockerfile

NiallJoeMaher committed:
Try a fix for Dockerfile

NiallJoeMaher committed:
Remove addDefaultCloudMapNamespace

NiallJoeMaher committed:
Bugfix/bump next (#383)

* Bump next version and revert dockerfile

* Add addDefaultCloudMapNamespace

NiallJoeMaher committed:
Update Node, update severSession function (#384)

NiallJoeMaher committed:
Undo changes suggested by Next team

NiallJoeMaher committed:
Remove hostname

NiallJoeMaher committed:
Add ssm permissions to tasks

NiallJoeMaher committed:
Bugfix/build issues (#385)

* Downgrade Next to 13.3

* Revert app-stack

NiallJoeMaher committed:
Try different version of Next/Superjson

NiallJoeMaher committed:
Testing a resizing function (#386)

NiallJoeMaher committed:
Remove unused og folder

NiallJoeMaher committed:
Add publishAssetsInParallel

NiallJoeMaher committed:
Add packages to lambdas

NiallJoeMaher committed:
Revert to .js

NiallJoeMaher committed:
Update lambda to nodefunction

NiallJoeMaher committed:
Add lambdas folders to ignore

NiallJoeMaher committed:
Change config for Lambda

NiallJoeMaher committed:
Moving back to node function

NiallJoeMaher committed:
Fix directory

NiallJoeMaher committed:
Add esbuild

NiallJoeMaher committed:
Add command hooks

NiallJoeMaher committed:
Add beforeBundling comands

NiallJoeMaher committed:
Remove command hooks

NiallJoeMaher committed:
Add some logging for debugging

NiallJoeMaher committed:
More logs for the log gods

NiallJoeMaher committed:
Move aws-cdk to dev deps

NiallJoeMaher committed:
Try change package location

NiallJoeMaher committed:
Test regex with image pattern

NiallJoeMaher committed:
Fix regex

NiallJoeMaher committed:
Update path

NiallJoeMaher committed:
Add Layers to Lambda

NiallJoeMaher committed:
Remove package location

NiallJoeMaher committed:
Add externalModules

NiallJoeMaher committed:
Try fix storage stack

NiallJoeMaher committed:
Try fix layer path

NiallJoeMaher committed:
Try bundling modules

NiallJoeMaher committed:
Try to create buffer

NiallJoeMaher committed:
Add putObjectCommand

NiallJoeMaher committed:
Remove logs, add target key

NiallJoeMaher committed:
Fit: cover

NiallJoeMaher committed:
Upgrade deps and update tsconfig (#387)

NiallJoeMaher committed:
Bump dependencies (#394)

NiallJoeMaher committed:
fix: upgrade next-auth from 4.22.3 to 4.22.4 (#395)

Snyk has created this PR to upgrade next-auth from 4.22.3 to 4.22.4.

See this package in npm:
https://www.npmjs.com/package/next-auth

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
fix: upgrade next-auth from 4.22.4 to 4.22.5 (#396)

Snyk has created this PR to upgrade next-auth from 4.22.4 to 4.22.5.

See this package in npm:
https://www.npmjs.com/package/next-auth

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
fix: upgrade next-auth from 4.22.5 to 4.23.0 (#397)

Snyk has created this PR to upgrade next-auth from 4.22.5 to 4.23.0.

See this package in npm:
https://www.npmjs.com/package/next-auth

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
fix: upgrade zod from 3.21.4 to 3.22.0 (#399)

Snyk has created this PR to upgrade zod from 3.21.4 to 3.22.0.

See this package in npm:
https://www.npmjs.com/package/zod

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
fix: upgrade @aws-sdk/client-ses from 3.390.0 to 3.391.0 (#400)

Snyk has created this PR to upgrade @aws-sdk/client-ses from 3.390.0 to 3.391.0.

See this package in npm:
https://www.npmjs.com/package/@aws-sdk/client-ses

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
fix: upgrade sharp from 0.32.4 to 0.32.5 (#405)

Snyk has created this PR to upgrade sharp from 0.32.4 to 0.32.5.

See this package in npm:
https://www.npmjs.com/package/sharp

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
fix: upgrade @headlessui/react from 1.7.16 to 1.7.17 (#406)

Snyk has created this PR to upgrade @headlessui/react from 1.7.16 to 1.7.17.

See this package in npm:
https://www.npmjs.com/package/@headlessui/react

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
fix: upgrade next-auth from 4.23.0 to 4.23.1 (#404)

Snyk has created this PR to upgrade next-auth from 4.23.0 to 4.23.1.

See this package in npm:
https://www.npmjs.com/package/next-auth

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
fix: upgrade aws-cdk-lib from 2.91.0 to 2.92.0 (#403)

Snyk has created this PR to upgrade aws-cdk-lib from 2.91.0 to 2.92.0.

See this package in npm:
https://www.npmjs.com/package/aws-cdk-lib

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/0c10c5ab-78ba-42aa-b499-d828ca03741c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
fix: upgrade @tanstack/react-query from 4.32.6 to 4.33.0 (#409)

Snyk has created this PR to upgrade @tanstack/react-query from 4.32.6 to 4.33.0.

See this package in npm:
https://www.npmjs.com/package/@tanstack/react-query

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
fix: upgrade @aws-sdk/client-ses from 3.391.0 to 3.395.0 (#408)

Snyk has created this PR to upgrade @aws-sdk/client-ses from 3.391.0 to 3.395.0.

See this package in npm:
https://www.npmjs.com/package/@aws-sdk/client-ses

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
fix: upgrade zod from 3.22.0 to 3.22.2 (#407)

Snyk has created this PR to upgrade zod from 3.22.0 to 3.22.2.

See this package in npm:
https://www.npmjs.com/package/zod

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
Update dependencies (#411)

NiallJoeMaher committed:
fix: upgrade react-textarea-autosize from 8.5.2 to 8.5.3 (#413)

Snyk has created this PR to upgrade react-textarea-autosize from 8.5.2 to 8.5.3.

See this package in npm:
https://www.npmjs.com/package/react-textarea-autosize

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
Update dependencies (#414)

NiallJoeMaher committed:
fix: upgrade @markdoc/markdoc from 0.3.0 to 0.3.1 (#415)

Snyk has created this PR to upgrade @markdoc/markdoc from 0.3.0 to 0.3.1.

See this package in npm:
https://www.npmjs.com/package/@markdoc/markdoc

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
Add more permissive license (#417)

NiallJoeMaher committed:
fix: upgrade @markdoc/markdoc from 0.3.1 to 0.3.2 (#418)

Snyk has created this PR to upgrade @markdoc/markdoc from 0.3.1 to 0.3.2.

See this package in npm:
https://www.npmjs.com/package/@markdoc/markdoc

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
Add Sentry setup (#420)

NiallJoeMaher committed:
fix: upgrade slate-react from 0.98.2 to 0.98.3 (#423)

Snyk has created this PR to upgrade slate-react from 0.98.2 to 0.98.3.

See this package in npm:
https://www.npmjs.com/package/slate-react

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

John-Paul-Larkin committed:
Fix bug #416 Random bookmarks (#427)

NiallJoeMaher committed:
fix: upgrade @tanem/react-nprogress from 5.0.48 to 5.0.49 (#428)

Snyk has created this PR to upgrade @tanem/react-nprogress from 5.0.48 to 5.0.49.

See this package in npm:
https://www.npmjs.com/package/@tanem/react-nprogress

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
fix: upgrade satori from 0.10.3 to 0.10.4 (#429)

Snyk has created this PR to upgrade satori from 0.10.3 to 0.10.4.

See this package in npm:
https://www.npmjs.com/package/satori

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
fix: upgrade react-hook-form from 7.45.4 to 7.46.0 (#430)

Snyk has created this PR to upgrade react-hook-form from 7.45.4 to 7.46.0.

See this package in npm:
https://www.npmjs.com/package/react-hook-form

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

NiallJoeMaher committed:
fix: upgrade react-hook-form from 7.46.0 to 7.46.1 (#431)

Snyk has created this PR to upgrade react-hook-form from 7.46.0 to 7.46.1.

See this package in npm:
https://www.npmjs.com/package/react-hook-form

See this project in Snyk:
https://app.snyk.io/org/nialljoemaher/project/8ac19f6c-c7f2-4720-acd1-09701979877c?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

John-Paul-Larkin committed:
create component and modal

John-Paul-Larkin committed:
add textarea for comment

John-Paul-Larkin committed:
add close on click outside

John-Paul-Larkin committed:
add toast on submit success

John-Paul-Larkin committed:
add email of commenter to report

NiallJoeMaher committed:
Feat/new editor starter (#424)

* WIP new editor starter from template
* Fix package-lock
* Remove BlobResult

John-Paul-Larkin committed:
add email to admin template

John-Paul-Larkin committed:
code refactor

John-Paul-Larkin committed:
addded reporters comment to the report email

DarrachBarneveld committed:
Theme (#442)

* Install next-theme, added basic theme toggler ui
* Created basic theme provider
* Updated tailwind config with anims and darkmode
* Added tailwind animations package
* Added @ ahead of tailwind animate plugin

Danishkar committed:
fix/changes twitter logo in footer to X (#444)

xiaoniuniu89 committed:
refactor(tip tap editor) and kill Slate

refactor(tip tap editor): 
* add support for editor integration 
* remove slate editor and dependencies 
* add support for rendering content from tip tap editor (#446)

Snowfall22 committed:
fix: change next version to 13.5.0 (#450)

* fix: update to 13.5.3

var-rishabh committed:
fix/changed Share to Twiter to X in articles (#452)

NiallJoeMaher committed:
Add first course chapter (intro to react) (#453)

xiaoniuniu89 committed:
Niu/gitpod setup script (#454)

* feat(gitpod setup script): add script to help hacktoberfest setup on gitpod

* update yaml instructions

* update sh script

* update .yaml order

* simplify setup script

* fix bug in auth callback url

John-Paul-Larkin committed:
Fishamble feature #393 article search (#449)

* Implement search feature
* made search bar responsive

pateldivyesh1323 committed:
feature: Add light mode to articles feed page (#445)

* Add light mode to articles feed page

NiallJoeMaher commented on server/trpc/router/comment.ts:
```diff
@@ -12,6 +12,30 @@ import {
+}
+[];
+
+// All I wanted from this file was to include user email in the ShapedResponse
```
I'll test this out and find out what's going on. 🤔

xiaoniuniu89 committed:
refactor(create page): move business logic into hook (#458)

NiallJoeMaher committed:
Bugfix/bump versions (#459)

* Bump Sharp and CDK

DarrachBarneveld committed:
Navbar (#461)

* Updated branch name
* Added navbutton styles light and dark mode, global css class
* Inverted colour on logo image for light mode
* Replaced toggle to render on un-auth navbar
* Minor style tweaks to themetoggle button

pateldivyesh1323 committed:
feat: Added Light mode to Profile Page (#460)

* Added Light mode to Profile Page

DarrachBarneveld committed:
Logins Theming, (#463)

* Update theme for login redirect page

pateldivyesh1323 committed:
feat/Add light mode to Footer Section (#465)

parinitapatil committed:
Updated icon color (#472)

Danishkar committed:
feat/adds the light mode for home page (#474)

AnuMessi10 committed:
Feat: Add Light Theme To Notifications Page (#480)

NiallJoeMaher commented:
@Fishamble you have any ideas why the build is failing or would you like me to pull the branch? I'll try get this merged asap.

John-Paul-Larkin commented:
Hi Niall,
Sorry about that. I think I know why its failing. I will try figure it out and get back to you soon.
When I click on the details of 'Vercel - deployment has failed' I get a 404. Is that expected? I was hoping for some error logs.

John-Paul-Larkin committed:
add api route to send email

John-Paul-Larkin commented:
I was trying to send email from the front end. I had watched a tutorial next with the App router, but did not understand how the pages router works. 
I have added a sendEmail api route, which I think will work.

NiallJoeMaher commented:
Aha! I see. I have API keys and things I can test against. 
Because we use trpc maybe I’ll open up another issue to help you out on this one. If I can wire up a utility or an example it might make your life easier. This is trickier than I thought. I might chat to you in Discord with some ideas. 🦾 This would be a fantastic addition.

Mrst12 committed:
Light mode for menu on [slug].tsx file (#466)

DarrachBarneveld committed:
Added custom hover and focus classes for social logos (#483)

* Added custom hover and focus classes for social logos

* Added global variable css in root and tailwind config

* Updated tailwing config with background image

NiallJoeMaher committed:
fix: package.json & package-lock.json to reduce vulnerabilities (#478)

The following vulnerabilities are fixed with an upgrade:
- https://snyk.io/vuln/SNYK-JS-POSTCSS-5926692

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

DarrachBarneveld committed:
Mobilenav - Refactor (#473)

* Refactored Mobilenav into seperate component, added usernavtypes into types folde

* Added dark mode text to hamburger

xiaoniuniu89 commented on utils/createReportEmailTemplate.ts:
```diff
@@ -0,0 +1,46 @@
+        <p>Details of reported comment:</p>
+        <p>link to post : <a style="color:yellow" href=${reportDetails.postLink ? reportDetails.postLink : ''}>${reportDetails.postLink ? reportDetails.postLink : ''}</a></p>
+        <p style='padding:3rem; border: 2px solid white;'>comment : ${reportDetails.reportedComment ? reportDetails.reportedComment : ''}</p>
+        <p>commment made by user : ${reportDetails.reportedOnName ? reportDetails.reportedOnName : ''} </p>
```
```suggestion
       <p>Comment made by user: {reportDetails?.reportedOnName || ''}</p>
```

xiaoniuniu89 replied:
for readability, this perhaps

xiaoniuniu89 commented on utils/createReportEmailTemplate.ts:
```diff
@@ -0,0 +1,46 @@
+interface ReportDetails {
+    reportedById: string | undefined;
```
```suggestion
    reportedById?: string;
```

xiaoniuniu89 commented on server/trpc/router/comment.ts:
```diff
@@ -12,6 +12,30 @@ import {
+interface ShapedResponse {
+  user: {
+    id: string;
+    username: string | null;
```
```suggestion
    username?: string;
```

xiaoniuniu89 commented on pages/api/sendEmail/sendEmail.ts:
```diff
@@ -0,0 +1,21 @@
+import type { NextApiRequest, NextApiResponse } from 'next';
+import sendEmail from '../../../utils/sendEmail';
+
+export default async function SendReportEmail(req: NextApiRequest, res: NextApiResponse) {
```
are we validating the incoming req here or anywhere else? for a valid req.body?

xiaoniuniu89 commented on pages/api/sendEmail/sendEmail.ts:
```diff
@@ -0,0 +1,21 @@
+    res.status(200).json({ success: true });
+  } catch (error) {
+    console.error('Error sending email:', error);
+    res.status(500).json({ success: false, error: 'Error sending email' });
```
```suggestion
    res.status(500).json({ success: false, error: `Error sending email: ${error}` });
```

will be nice here too

xiaoniuniu89 commented on components/Comments/ReportPost.tsx:
```diff
@@ -0,0 +1,145 @@
+  };
+
+  const handleSubmit = async (e:React.FormEvent) =>{
+    e.preventDefault();
```
can we validate session here before running the logic? Will a user who is not logged in be able to send emails?

NiallJoeMaher replied:
Agreed! Will stop spam abuse hopefully.

John-Paul-Larkin replied:
Hi. I validate the session when the user clicks the flag. I thought that would do the job,  but I can add it here too.

xiaoniuniu89 commented on components/Comments/ReportPost.tsx:
```diff
@@ -0,0 +1,145 @@
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          recipient: process.env.ADMIN_EMAIL || '',
```
I am wondering if we can move this into an api route and have the client side logic call the api route instead? this way client is not accessing env variables

John-Paul-Larkin replied:
I knew this was not right. The problem is I dont have a good understanding of the pages router and trpc. Im going to take a few days to get up to speed on them and return to this with a better understanding.

xiaoniuniu89 commented on components/Comments/ReportPost.tsx:
```diff
@@ -0,0 +1,145 @@
+      });
+      toast.success('Report sent');
+    } catch (error) {
+      console.log('Error attempting to email report', error);
```
do we need the console.logs and console.errors? they are not providing error handling and you usually have a second call to return a toast with the error.

John-Paul-Larkin replied:
You are right. Force of habit there :)

xiaoniuniu89 commented:
good job with this!

John-Paul-Larkin commented:
Hi Daniel. 
Ill work on it and get back with the changes.
I appreciate all the feedback. Thanks.

NiallJoeMaher committed:
Remove forceDockerBundling

NiallJoeMaher committed:
Add dockerEnabled for SelfMutation and synth

John-Paul-Larkin committed:
Merge branch 'codu-code:develop' into Fishamble-feature-#168-report-posts/comments

John-Paul-Larkin committed:
fix readability issues

John-Paul-Larkin committed:
add route to send email

John-Paul-Larkin committed:
Merge remote-tracking branch 'refs/remotes/origin/Fishamble-feature-#168-report-posts/comments' into Fishamble-feature-#168-report-posts/comments

John-Paul-Larkin committed:
move ADMIN_EMAIL env to sendEmail function

John-Paul-Larkin commented on components/Comments/ReportPost.tsx:
```diff
@@ -15,6 +16,11 @@ interface Props {
+  const { mutate:sendEmail }  = trpc.sendEmail.send.useMutation({ 
+      onSuccess: () => { toast.success('Report sent')},
+      onError: () => {  toast.error('Oops, something went wrong, please send us a message on discord https://github.com/codu-code/codu');  }
+  })
```
I have an issue here.  The user will always recieve the success toast. Any error with sending the email is handled by nodemailer, and I dont know how to throw it here on the front end. I experimented with try catch errors, but could not get it to work.

John-Paul-Larkin replied:
I created a route and moved the sendEmail function to the backend. Hopefully the email will now work.

xiaoniuniu89 commented on server/trpc/router/sendEmail.ts:
```diff
@@ -0,0 +1,19 @@
+import toast from "react-hot-toast";
+import { SendEmailSchema } from "../../../schema/comment";
+import sendEmail from "../../../utils/sendEmail";
```
this may need to be updated to fix your issue propagating error from nodemailer back to the client - we should return a promise so we can Propagate the error back to the front end. in sendEmail wrap the callback function transporter.sendMail in a promise. 
```
return new Promise((resolve, reject) => {
    transporter.sendMail( ..... . etc 
```
you also need to add some logic for reject so it will pass back to the client:
`if(err){reject(new Error("Failed to send email"))}`

xiaoniuniu89 replied:
try using try catch down below in sendEmailRouter and if it doesnt work wrap sendEmail in utils/sendEmail in a promise and see if it will propogate to the client

xiaoniuniu89 commented on server/trpc/router/sendEmail.ts:
```diff
@@ -0,0 +1,19 @@
+
+export const sendEmailRouter = router({
+
+    send:protectedProcedure
```
you can do your try catch here and in the catch 

catch (error) {
        throw new Error("Failed to send email");
        
        that should trigger the onError for your toast

xiaoniuniu89 commented on server/trpc/router/sendEmail.ts:
```diff
@@ -0,0 +1,19 @@
+        .input(SendEmailSchema)
+        .mutation(async ({input}) => {
+            const { recipient, htmlMessage, subject } = input;
+            sendEmail({
```
```
try {
  await sendEmail({
      recipient: process.env.ADMIN_EMAIL || '',
      htmlMessage,
      subject,
  });
} catch (error) {
  throw new Error(`Failed to send email: ${error}`);
}
```

xiaoniuniu89 replied:
try this to see if it trigger onError

John-Paul-Larkin replied:
I tried this exact thing, but it did not work. I also tried the same on the sendEmail function in /utils/sendEmail, but that did not work either. I will experiment more with wrapping a promise.
Thanks for all the feedback. Its very helpful.

John-Paul-Larkin committed:
wrap sendEmail in promise

John-Paul-Larkin commented:
As suggested, I wrapped utils/sendEmail in a promise, which on reject propagates to the client. I had to modify the function, but it is only used in one other place, the createUser function in api/auth/[...nextauth].ts. 
createUser is already wrapped in a try..catch.
Hopefully we can finally get this closed. Thanks again for the help Daniel.

xiaoniuniu89 commented on server/trpc/router/sendEmail.ts:
```diff
@@ -0,0 +1,24 @@
+                });
+              } catch (error) {
+                console.log(`Failed to send email: ${error}`)
+                throw new Error();
```
```suggestion
                throw new Error(error);
```

John-Paul-Larkin replied:
I had it like that because I didnt want to pass the error to the client. This way the error is thrown, triggering the error toast, and the error message is logged on the server side, not the browser.

NiallJoeMaher commented:
I think this branch has more conflicts than I've ever seen because of some of the work I done behind the scenes 🙈 Sorry about that! If you don't mind (and you can remember), I'm about to finish a migration to the `/app` router in Next, and then we can apply the changes and get them moved. It might be easier on a new branch though because this one looks scary 😹 

Sorry for the chaos 😢

John-Paul-Larkin commented:
> I think this branch has more conflicts than I've ever seen because of some of the work I done behind the scenes 🙈 Sorry about that! If you don't mind (and you can remember), I'm about to finish a migration to the `/app` router in Next, and then we can apply the changes and get them moved. It might be easier on a new branch though because this one looks scary 😹
> 
> Sorry for the chaos 😢

Honestly, I have no clue what was going on here with all these confilcts. :) 
The actual changes are confined to just a few modules, so I will close this and start fresh with a new branch and pull request.
